src/components/IstioWizards/IstioWizard.tsx

```diff
@@ -166,18 +171,59 @@ class IstioWizard extends React.Component<Props, State> {
                   host: this.props.serviceName,
                   subset: wkdNameVersion[workload.name]
                 },
-                weight: workload.traffic
+                weight: workload.weight
               };
             })
           }
         ]
+      };
+    }
+
+    if (this.props.type === WIZARD_MATCHING_ROUTING) {
+      // VirtualService from the routes
+      wizardVS.spec = {
+        hosts: [this.props.serviceName],
+        http: this.state.rules.map(rule => {
+          const httpRoute: HTTPRoute = {};
+          const destW: DestinationWeight = {
+            destination: {
+              host: this.props.serviceName
+            }
+          };
+          if (rule.routeType === ROUTE_TYPE.WORKLOAD) {
+            destW.destination.subset = wkdNameVersion[rule.route];
+          }
+          httpRoute.route = [destW];
+          if (rule.matches.length > 0) {
+            httpRoute.match = this.buildHTTPMatchRequest(rule.matches);
+          }
+          return httpRoute;
+        })
+      };
+    }
+
+    if (this.state.mtlsMode !== NONE || this.state.loadBalancer !== NONE) {
+      wizardDR.spec.trafficPolicy = {};
+      if (this.state.mtlsMode !== NONE) {
+        wizardDR.spec.trafficPolicy.tls = {
+          mode: this.state.mtlsMode
+        };
       }
-    };
+      if (this.state.loadBalancer !== NONE) {
+        wizardDR.spec.trafficPolicy.loadBalancer = {
+          simple: this.state.loadBalancer
+        };
+      }
+    }
+
     return [wizardDR, wizardVS];
```

Comment:
This could be

```ts
wizardDR.spec.trafficPolicy = {};
if (this.state.mtlsMode !== NONE) {
  wizardDR.spec.trafficPolicy.tls = {
          mode: this.state.mtlsMode
        };
}
if (this.state.loadBalancer !== NONE) {
  wizardDR.spec.trafficPolicy.loadBalancer = {
          simple: this.state.loadBalancer
        };
}
```

or we can't we have the trafficPolicy empty ? the proposal is avoid in the worst of the cases 3 conditionals.

Reply:
Yeah, the logic is to not have the trafficPolicy in the object.
With your suggestion, it simplifies the code but the 'trafficPolicy' will be always present.

Reply:
ok